provider/pkg/provider/provider.go

```diff
@@ -544,13 +589,13 @@ func (p *googleCloudProvider) waitForResourceOpCompletion(
 			logging.V(9).Info("Getting self link URL")
 			sl, err := getKNativeSelfLinkURL(resp)
 			if err != nil {
-				return nil, err
+				return resp, err
```

Comment:
Its important to return any partial state as part of the creation to allow checkpointing.

Reply:
Let's add a comment with that info to the code as well

Reply:
Updated the docstring for the method